### ISSUE TITLE
Support "Description" for sources without descriptions

### ISSUE THREAD

**Issue by pleia2** [openmainframeproject/software-discovery-tool#45]
In #42 we got support for a description column :tada: but now when we try to import JSON sources that do not include the Description column (including all of the legacy PDS sources, which will never include a Description column), an error is thrown:

```
[Tue Jul 06 17:55:37.944969 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx] mod_wsgi (pid=1720): Failed to exec Python script file '/var/www/sdt.princessleia.com/htdocs/src/SDT.wsgi'.
[Tue Jul 06 17:55:37.945150 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx] mod_wsgi (pid=1720): Exception occurred processing WSGI script '/var/www/sdt.princessleia.com/htdocs/src/SDT.wsgi'.
[Tue Jul 06 17:55:37.946432 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx] Traceback (most recent call last):
[Tue Jul 06 17:55:37.946510 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]   File "/var/www/sdt.princessleia.com/htdocs/src/classes/package_search.py", line 54, in loadPackageData
[Tue Jul 06 17:55:37.946518 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]     json_data = json.load(open(distro_data_file))
[Tue Jul 06 17:55:37.946528 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]   File "/usr/lib/python3.7/json/__init__.py", line 296, in load
[Tue Jul 06 17:55:37.946533 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]     parse_constant=parse_constant, object_pairs_hook=object_pairs_hook, **kw)
[Tue Jul 06 17:55:37.946542 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]   File "/usr/lib/python3.7/json/__init__.py", line 348, in loads
[Tue Jul 06 17:55:37.946547 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]     return _default_decoder.decode(s)
[Tue Jul 06 17:55:37.946554 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]   File "/usr/lib/python3.7/json/decoder.py", line 337, in decode
[Tue Jul 06 17:55:37.946559 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
[Tue Jul 06 17:55:37.946570 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]   File "/usr/lib/python3.7/json/decoder.py", line 355, in raw_decode
[Tue Jul 06 17:55:37.946578 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]     raise JSONDecodeError("Expecting value", s, err.value) from None
[Tue Jul 06 17:55:37.946609 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx] json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
[Tue Jul 06 17:55:37.946647 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]
[Tue Jul 06 17:55:37.946653 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx] During handling of the above exception, another exception occurred:
[Tue Jul 06 17:55:37.946658 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]
[Tue Jul 06 17:55:37.946666 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx] Traceback (most recent call last):
[Tue Jul 06 17:55:37.946709 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]   File "/var/www/sdt.princessleia.com/htdocs/src/SDT.wsgi", line 5, in <module>
[Tue Jul 06 17:55:37.946715 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]     from main import app as application
[Tue Jul 06 17:55:37.946721 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]   File "/var/www/sdt.princessleia.com/htdocs/src/main.py", line 11, in <module>
[Tue Jul 06 17:55:37.946724 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]     package_search = PackageSearch.load()
[Tue Jul 06 17:55:37.946732 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]   File "/var/www/sdt.princessleia.com/htdocs/src/classes/package_search.py", line 132, in load
[Tue Jul 06 17:55:37.946736 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]     return cls.get_instance()
[Tue Jul 06 17:55:37.946743 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]   File "/var/www/sdt.princessleia.com/htdocs/src/classes/package_search.py", line 123, in get_instance
[Tue Jul 06 17:55:37.946747 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]     cls.INSTANCE.package_data = cls.loadPackageData()
[Tue Jul 06 17:55:37.946754 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]   File "/var/www/sdt.princessleia.com/htdocs/src/classes/package_search.py", line 58, in loadPackageData
[Tue Jul 06 17:55:37.946759 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]     json_data = cls.preparePackageData()
[Tue Jul 06 17:55:37.946766 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]   File "/var/www/sdt.princessleia.com/htdocs/src/classes/package_search.py", line 93, in preparePackageData
[Tue Jul 06 17:55:37.946769 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx]     cachedPackage["D"] = pkg["description"]
[Tue Jul 06 17:55:37.946784 2021] [wsgi:error] [pid 1720] [remote xxxx:xxxx] KeyError: 'description'
```

**Comment by indranilm256** [comment]
Taking up this issue.

**Comment by indranilm256** [comment]
Hello @pleia2 could you please help with the data file and the steps required to reproduce the error. Thanks.

**Comment by rachejazz** [comment]
@indranilm256 i think it happens when the json files lack a "description" key. So..... try this maybe:
```
> ./bin/package_build <any existing json file name from PDS with .json extension>
> rm ./distro_data/distro_data/cached.json
# restart server
```

**Comment by rachejazz** [comment]
There are 2 ways of resolving this, 
1. Keep description key optional in the json
2. Or just write another script to format PDS data fit for SDT by adding description column automatically :'D

2nd option will be easier 1st is the right way to do it

**Comment by indranilm256** [comment]
yeah I'd be leaning towards the 1st one, what we need to do is check if the description key is present in the json object and then read its value. I suppose that should should be an easy fix :)

**Comment by indranilm256** [comment]
> @indranilm256 i think it happens when the json files lack a "description" key. So..... try this maybe:
> 
> ```
> > ./bin/package_build <any existing json file name from PDS with .json extension>
> > rm ./distro_data/distro_data/cached.json
> # restart server
> ```

Thanks

**Comment by pleia2** [comment]
Thanks @rachejazz! @indranilm256 let me know if you need more precise steps to replicate and I'll provide them.

**Comment by indranilm256** [comment]
@pleia2 I was able to replicate the issue. Thanks

**Comment by pleia2** [comment]
Fixed in #53
